samples/attestation_plugin/enclave/plugin1/plugin1.c

```diff
@@ -0,0 +1,171 @@
+    return 0;
+}
+
+static int oe_on_unregister(
+    oe_quote_customization_plugin_context_t* plugin_context)
+{
+    OE_UNUSED(plugin_context);
+    return 0;
+}
+
+static int _calc_sha256(
+    const uint8_t* user_data,
+    size_t user_data_size,
+    uint8_t* output)
+{
+    mbedtls_sha256_context ctx;
+    mbedtls_sha256_init(&ctx);
+    int ret = -1;
+
+    if (mbedtls_sha256_starts_ret(&ctx, 0) != 0)
+        goto done;
+
+    if (mbedtls_sha256_update_ret(&ctx, user_data, user_data_size) != 0)
+        goto done;
+
+    if (mbedtls_sha256_finish_ret(&ctx, output) != 0)
+        goto done;
+
+    ret = 0;
+
+done:
+    mbedtls_sha256_free(&ctx);
+    return ret;
+}
+
+static int oe_get_evidence(
+    oe_quote_customization_plugin_context_t* plugin_context,
+    const uint8_t* user_data,
+    size_t user_data_size,
+    uint8_t** evidence_buffer,
+    size_t* evidence_buffer_size)
+{
+    // Need to hash the user data first.
+    uint8_t* report_data = NULL;
+    size_t report_data_size = 0;
+    uint8_t hash[32];
+    int ret = -1;
+    oe_result_t oe_ret;
+    uint8_t* report_buffer = NULL;
+    size_t report_buffer_size = 0;
+    size_t total_evidence_size = 0;
+
+    if (user_data)
+    {
+        if (_calc_sha256(user_data, user_data_size, hash) != 0)
+            goto done;
+
+        report_data = hash;
+        report_data_size = sizeof(hash);
+    }
+
+    // Technically, this will wrap the sgx report with the OE header,
+    // which will get wrapped again in the caller of this function,
+    // but that's fine.
+    oe_ret = oe_get_report(
+        OE_REPORT_FLAGS_REMOTE_ATTESTATION,
+        report_data,
+        report_data_size,
+        NULL,
+        0,
+        &report_buffer,
+        &report_buffer_size);
+
+    if (oe_ret != OE_OK)
+        goto done;
+
+    if (!user_data)
+    {
+        ret = 0;
+        goto done;
+    }
+
+    // Now, we need to append the report data to the end of the report.
+    oe_ret = oe_safe_add_sizet(
+        report_data_size, report_buffer_size, &total_evidence_size);
+    if (oe_ret != OE_OK)
+        goto done;
+
+    *evidence_buffer = (uint8_t*)malloc(total_evidence_size);
+    if (*evidence_buffer == NULL)
+        goto done;
+
+    memcpy(*evidence_buffer, report_buffer, report_buffer_size);
+    memcpy(
+        *evidence_buffer + report_buffer_size, report_data, report_data_size);
+    *evidence_buffer_size = total_evidence_size;
+
+    ret = 0;
+
+done:
+    return ret;
+}
+
+static int oe_free_evidence(
+    oe_quote_customization_plugin_context_t* plugin_context,
+    uint8_t* evidence_buffer)
+{
+    oe_free_report(evidence_buffer);
+    return 0;
+}
+
+static int oe_verify_evidence(
+    oe_quote_customization_plugin_context_t* plugin_context,
+    const uint8_t* evidence_buffer,
+    size_t evidence_buffer_size,
+    oe_claim_element_t** claims,
+    size_t* claim_count,
+    uint8_t** user_data,
+    size_t* user_data_size)
+{
+    return 0;
```

Comment:
It would be great to do something here to make this sample more interesting.

Reply:
Yeah, I haven't finished sample 1 yet, since we had some discussions about attestation this week. I'm was planning on it do the verification like how the report generation is implemented.